base_layer/core/src/consensus/chain_strength_comparer.rs

```diff
@@ -0,0 +1,79 @@
+use crate::blocks::BlockHeader;
+use std::{cmp::Ordering, fmt::Debug};
+
+pub trait ChainStrengthComparer: Debug {
+    fn compare(&self, a: &BlockHeader, b: &BlockHeader) -> Ordering;
+}
+
+#[derive(Default, Debug)]
+pub struct AccumulatedDifficultySquaredComparer {}
+
+impl ChainStrengthComparer for AccumulatedDifficultySquaredComparer {
+    fn compare(&self, a: &BlockHeader, b: &BlockHeader) -> Ordering {
+        let a_val = a.total_accumulated_difficulty_inclusive_squared();
+        let b_val = b.total_accumulated_difficulty_inclusive_squared();
+        if a_val < b_val {
+            Ordering::Less
+        } else {
+            // f64's can never really be equal, so there is no `cmp` for f64
+            // there are also weird NaN edge cases
+            Ordering::Greater
+        }
+    }
```

Comment:
Might want to use `total_cmp`
https://doc.rust-lang.org/beta/std/primitive.f64.html#method.total_cmp
 since the comparison is part of a floating point standard.